scripts/src/utils/RateLimit.ts

```diff
@@ -0,0 +1,53 @@
+import * as moment from "moment";
+import { getRedisClient, RedisAsyncClient } from "../redis";
+
+export class RateLimit {
+	private bucketPrefix: string;
+	private rateLimitValue: number;
+	private redis: RedisAsyncClient;
+	private windowSize: number;
+	private ttl: number;
+
+	constructor(bucketPrefix: string, rateLimitValue: number, windowSizeMomentObject: moment.Duration) {
+		this.bucketPrefix = bucketPrefix;
+		this.rateLimitValue = rateLimitValue;
+		this.windowSize = 0;
+		this.ttl = 0;
+		this.setWindowSizeAndTTL(moment.duration(windowSizeMomentObject));
+		this.redis = getRedisClient();
+	}
+
+	public async checkRate(): Promise<boolean> {
+		return await this.check(1);
+	}
+
+	public async checkAmount(amount: number): Promise<boolean> {
+		return await this.check(amount);
+	}
+
+	private async check(step: number) {
```

Comment:
Why not:
```
public async check(step?: number) {
    if (step === undefined) {
        step = 1;
    }
}
```
?

Reply:
For semantic reasons only: separate `checkRate` and `checkAmount` names are more clear, inside it encapsulates the generic `check` method call

Reply:
i'm not a fan of this approach, if they are doing the same thing then they should be the same.
but i don't care that much

Reply:
That's the point: they don't do same things, in the code it looks similar but the purpose is different